pkg/cache/cache_from_test.go

```diff
@@ -0,0 +1,217 @@
+	cf := CacheFrom{}
+	expected := CacheFrom{"okteto.dev/test-account/test-image:cache"}
+
+	imageCtrl := &mockImageCtrl{}
+	registry := &mockRegistryWithError{imageCtrl: imageCtrl}
+
+	cf.AddDefaultPullCache(registry, image)
+
+	assert.Equal(t, cf, expected)
+}
+
+func Test_hasCacheFromImage(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		cf       CacheFrom
+		expected bool
+	}{
+		{
+			name:     "not found",
+			input:    "test-registry/test-image:cache",
+			cf:       CacheFrom{"test-registry/test-image:test-tag"},
+			expected: false,
+		},
+		{
+			name:     "found",
+			input:    "test-registry/test-image:cache",
+			cf:       CacheFrom{"test-registry/test-image:cache"},
+			expected: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := tt.cf.hasCacheFromImage(tt.input)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func Test_UnmarshalYAML(t *testing.T) {
+	tests := []struct {
+		name     string
+		data     []byte
+		expected interface{}
+	}{
+		{
+			name:     "empty",
+			data:     []byte("[]"),
+			expected: CacheFrom{},
+		},
+		{
+			name:     "one single item",
+			data:     []byte("test-registry/test-image:cache"),
+			expected: CacheFrom{"test-registry/test-image:cache"},
+		},
+		{
+			name:     "one item as list",
+			data:     []byte(`["test-registry/test-image:cache"]`),
+			expected: CacheFrom{"test-registry/test-image:cache"},
+		},
+		{
+			name:     "one item as list",
+			data:     []byte(`["test-registry/test-image:cache"]`),
+			expected: CacheFrom{"test-registry/test-image:cache"},
+		},
+		{
+			name: "two items",
+			data: []byte(`
+- test-registry/test-image:1.0.0
+- test-registry/another-image:1.0.0`),
+			expected: CacheFrom{"test-registry/test-image:1.0.0", "test-registry/another-image:1.0.0"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var cf CacheFrom
+			err := yaml.Unmarshal([]byte(tt.data), &cf)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.expected, cf)
+		})
+	}
+}
+
+func Test_UnmarshalYAML_WithError(t *testing.T) {
+	var cf CacheFrom
+
+	err := cf.UnmarshalYAML(func(interface{}) error {
+		return assert.AnError
+	})
+
+	assert.ErrorIs(t, err, assert.AnError)
+}
+
+func Test_MarshalYAML(t *testing.T) {
+	tests := []struct {
+		name     string
+		cf       CacheFrom
+		expected string
+	}{
+		{
+			name:     "empty",
+			cf:       CacheFrom{},
+			expected: "[]\n",
+		},
+		{
+			name:     "one image",
+			cf:       CacheFrom{"test-registry/test-image:cache"},
+			expected: "- test-registry/test-image:cache\n",
+		},
+		{
+			name:     "two images",
+			cf:       CacheFrom{"test-registry/test-image-1:cache", "test-registry/test-image-2:cache"},
+			expected: "- test-registry/test-image-1:cache\n- test-registry/test-image-2:cache\n",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result, err := yaml.Marshal(tt.cf)
```

Comment:
```suggestion
			result, err := yaml.Marshal(&tt.cf)
```
I think it was the problem to not reach `Marshal()` func. The signature that we use didn't fit with the real one.

Reply:
good catch @AdrianPedriza, I've updated the code! 🙏